helper/winsvc/service_windows.go

```diff
@@ -1,46 +1,85 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: BUSL-1.1
 
-//go:build windows
-// +build windows
-
 package winsvc
 
 import (
-	wsvc "golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/eventlog"
 )
 
-type serviceWindows struct{}
+// Commands that are currently supported
+const SERVICE_ACCEPTED_COMMANDS = svc.AcceptStop | svc.AcceptShutdown
+
+type serviceWindows struct {
+	evtLog Eventlog
+}
 
 func init() {
-	interactive, err := wsvc.IsAnInteractiveSession()
+	isSvc, err := svc.IsWindowsService()
 	if err != nil {
 		panic(err)
 	}
-	// Cannot run as a service when running interactively
-	if interactive {
+	// This should only run when running
+	// as a service
+	if !isSvc {
 		return
 	}
-	go wsvc.Run("", serviceWindows{})
+
+	go executeWindowsService()
 }
 
 // Execute implements the Windows service Handler type. It will be
 // called at the start of the service, and the service will exit
 // once Execute completes.
-func (serviceWindows) Execute(args []string, r <-chan wsvc.ChangeRequest, s chan<- wsvc.Status) (svcSpecificEC bool, exitCode uint32) {
-	const accCommands = wsvc.AcceptStop | wsvc.AcceptShutdown
-	s <- wsvc.Status{State: wsvc.Running, Accepts: accCommands}
+func (srv serviceWindows) Execute(args []string, r <-chan svc.ChangeRequest, s chan<- svc.Status) (svcSpecificEC bool, exitCode uint32) {
+	const accCommands = svc.AcceptStop | svc.AcceptShutdown
```

Comment:
I think this `accCommands` line is vestigial with the addition of `SERVICE_ACCEPTED_COMMANDS`